pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -180,12 +181,24 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 				buildRun.Status.Reason = lastTaskRun.Status.Conditions[0].Reason
 			}
 		}
-		buildRun.Status.LatestTaskRunRef = &lastTaskRun.Name
-		buildRun.Status.StartTime = lastTaskRun.Status.StartTime
-		buildRun.Status.CompletionTime = lastTaskRun.Status.CompletionTime
 		if buildRun.Status.BuildSpec == nil {
 			buildRun.Status.BuildSpec = &build.Spec
 		}
+
+		if lastTaskRun.Status.CompletionTime != nil && buildRun.Status.CompletionTime == nil {
+			buildRun.Status.CompletionTime = lastTaskRun.Status.CompletionTime
+
+			// Increase BuildRun count in metrics
+			buildmetrics.BuildRunCountInc(build.Spec.StrategyRef.Name)
```

Comment:
Hereby, you are only counting build runs that are complete. Any build run that fails on creation of the task run will be missed. Same suggestion as for the build count: the predicate function imo works best.

Reply:
Yep, like my comments above, if we all think `predicate` is the better way at current stage, let us move to that one, I also think it is a better way to cover both succeed and fail cases.

But we also need to think the about the case after the controller is restarted and metric is cleaned.